Source/AliveLib/Input.cpp

```diff
@@ -0,0 +1,39 @@
+#include "stdafx.h"
+#include "Input.hpp"
+#include <windows.h>
+#include "logger.hpp"
+#include "Function.hpp"
+#include "easylogging++.h"
+
+// -- Variables -- //
+
+ALIVE_ARY(1, 0xBD2F60, char, 256, sInputKeyStates_BD2F60, {});
+
+ALIVE_VAR(1, 0x5C2EF4, int, sJoystickEnabled_5C2EF4, 0);
+ALIVE_VAR(1, 0x5C2EFC, int, sJoystickNumButtons_5C2EFC, 0);
+ALIVE_VAR(1, 0x5C2F00, int, sJoystickID_5C2F00, 0);
+
+// -- Functions -- //
+
+// Zeros the input key state array.
+void CC Input_InitKeyStateArray_4EDD60()
+{
+	memset(sInputKeyStates_BD2F60, 0, 256u);
+}
+ALIVE_FUNC_IMPLEX(0x4EDD60, Input_InitKeyStateArray_4EDD60, true);
```

Comment:
For `true` I normally have a file level #define INPUT_IMPL true so that you can change it to false when trying to track down regressions